aim/api/infra.py

```diff
@@ -154,3 +155,46 @@ def is_available(self, context):
             LOG.debug("APIC %s is not available. Last update time was %s" %
                       (self.apic_host, self.last_update_timestamp))
             return False
+
+
+# REVISIT(kentwu): We will need to deprecate this once there is
+# a proper fix in the openShift IPI installer.
+class NestedParameter(resource.ResourceBase):
+    """Nested parameters needed for openShift on openStack install"""
+
+    identity_attributes = t.identity(
+        ('tenant_id', t.name),
```

Comment:
What is tenant_id here? I notice this is not tenant_name, which is generally an ACI tenant name. If this is supposed to be an OpenStack project_id, which I suspect based on the GBP repo patch (similar comment coming there), I'd strongly prefer you call it project_id rather than tenant_id. The term tenant has been deprecated in OpenStack for years, and in AIM, project_id is a lot less likely to be confused with tenant_name.

Reply:
ok will change it to project_id then.